tests/test_biologic_ec_measurement.py

```diff
@@ -0,0 +1,158 @@
+from pathlib import Path
+import time
+from ixdat import Measurement
+from ixdat.db import change_database
+
+#  If tox crashes when trying to import matplotlib, see:
+#    https://github.com/ixdat/ixdat/issues/10
+
+
+path_to_file = Path(__file__).parent.parent / "test_data/biologic/Pt_poly_cv.mpt"
+change_database("directory", project_name="test_biologic_ec_measurement")
+
+
+class TestLog:
+    def __init__(self, path_to_file="~/ixdat_test_output.txt"):
+        self.path_to_file = Path(path_to_file).expanduser()
+        with open(self.path_to_file, "w") as f:
+            f.write("\n" + "-" * 80 + "\n")
+            f.write(f"RUNNING MODULE {__file__} AT t={time.time()}\n")
+            f.write("-" * 80 + "\n")
+
+    def print(self, s):
+        with open(self.path_to_file, "a") as f:
+            f.write(s + "\n")
+
+
+log = TestLog()
+log.print(f"cwd = {Path('.').absolute().resolve()}")
+
+
+def test_append_essential_series():
+    meas = Measurement.read(path_to_file, reader="biologic")
+    log.print(f"successfully read {meas}")
+
+    assert meas.technique == "EC"
+    assert len(meas["potential"].data) == len(meas["time/s"].data)
+
+    # Calibrate the potential and check that it works.
+    meas.calibrate_RE(RE_vs_RHE=1)
+    assert meas.v[0] - meas["raw_potential"].data[0] == meas.RE_vs_RHE
+
+    # To make it complex, we first select a couple cycles, this time by converting to
+    # a cyclic voltammagram and indexing, and append them:
+    cv = meas.as_cv()
+    cvs_1_plus_2 = cv[1] + cv[2]
+
+    # Check that the calibration survived all that:
+    assert cvs_1_plus_2.RE_vs_RHE == meas.RE_vs_RHE
+    # Check that the main time variable, corresponding to potential, wasn't corrupted:
+    assert len(cvs_1_plus_2.grab("potential")[0]) == len(cvs_1_plus_2["time/s"].data)
+    # Check that the selector is still available and works with the main time variable:
+    assert len(cvs_1_plus_2.selector.data) == len(cvs_1_plus_2.t)
+    # ^ perfect! Tests that don't work now but should work after this restructure :D
+
+
+def test_save_load():
+    meas = Measurement.read(path_to_file, reader="biologic")
+    log.print(f"successfully read {meas}")
+
+    # To make it complex, we first select a couple cycles, this time with select(),
+    # and append them:
+    meas1 = meas.select(cycle=1)
+    meas2 = meas.select(cycle=3)
+    meas12 = meas1 + meas2
+
+    # Save and load the series:
+    i = meas12.save()
+    meas12_loaded = Measurement.get(i)
+
+    # First, check that we have the full time vector and no more:
+    assert len(meas12.t) == len(meas12_loaded.t)
+
+    # Now let's try calibrating just the loaded one:
+    RE_vs_RHE = 1
+    A_el = 2
+    meas12_loaded.calibrate(RE_vs_RHE=RE_vs_RHE, A_el=A_el)
+
+    # First we check that its v attribute is calibrated by comparing to the grabbed
+    #   uncalibrated potential:
+    assert meas12_loaded.v[0] == meas12.grab("raw_potential")[1][0] + RE_vs_RHE
```

Comment:
This line here though is indirectly testing save and load! The assertion only passes if RE_vs_RHE survives save+load and is in the right place in meas12_loaded

Reply:
Put in as a TODO.
@KennethNielsen , there is so much I don't understand about software testing. Let's take it seperate from the rest? 
These functional tests were invaluable as a development tool, but I have no idea how to use pytest or write good software tests. I think it would be most efficient to do so together as part of a later PR.

Reply:
I'm not sure I know how to write good software tests :D Seriously, I find myself regularly stumble at how to do it really good. Like e.g. the whole subject of testing functionality without making the tests too fragile is really hard. In any case, I do know a little and I think it is a good suggestion that I do an iteration on the existing tests once this is merged. Do note however, that I am not pointing fingers at this. I have often seen comments from cleverer people than me, highlighting how just a few well placed functional tests can go a long way towards preventing regressions. I think that is what you have had here, and the suggestions I made is more about the form they are written in. In any case, it would make a good assignment for me to do a PR with an iteration. Can I persuade you to add a link to this discussion to the comment?

Reply:
link added.